### PR TITLE
[Feature] Add endpoint command, refactor Makefile to cross build target

### PR DESCRIPTION
**Pull request by kezhenxu94**

### CODE REVIEW

File: commands/metrics/linear/linear-metrics.go [hunk 1]
```diff
@@ -50,8 +55,15 @@ var Command = cli.Command{
 		step := ctx.Generic("step")
 		metricsName := ctx.String("name")
 
+		var id *string = nil
+
+		if serviceID := ctx.String("id"); serviceID != "" {
+			id = &serviceID
+		}
```
Comment (wu-sheng) [inline commands/metrics/linear/linear-metrics.go]: linear metrics can work will service, instance and endpoint too. I think this is not wrong here.

File: commands/metrics/single/single-metrics.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		flags.DurationFlags,
+		[]cli.Flag{
+			cli.StringFlag{
+				Name:     "name",
+				Usage:    "metrics `NAME`, which should be defined in OAL script",
+				Required: true,
+			},
+			cli.StringSliceFlag{
+				Name:     "ids",
+				Usage:    "`IDs`, IDs required by the metric type",
```
Comment (wu-sheng) [inline commands/metrics/single/single-metrics.go]: IDs are available in all `ls` command, right?

**Comment by wu-sheng** [comment]
Based on the discussion between @kezhenxu94 and me, we are going to use ID as parameter for metrics query. Then leverage the Linux command, we could do this
```shell
> ./bin/swctl instance ls --service-id 3 | jq '.[].id' | xargs ./bin/swctl --debug single-metrics --name service_instance_sla --ids
```

Currently, let's add document for this complex command. Re-organize the doc, then we are closing to 0.1.0 release. 

@kezhenxu94 After this, let's work on LICENSE and require test in ML. Anything miss?

File: README.md [hunk 3]
```diff
@@ -169,5 +169,98 @@ This section covers all the available commands in SkyWalking CLI and their usage
...
+```
+
+</details>
+
+<details>
+
+<summary>Query a single metrics value for a specific endpoint</summary>
+
+```shell
+$ ./bin/swctl endpoint ls --service-id 3 | jq '.[] | select(.name == "Kafka/test-trace-topic/Consumer/test").id' | xargs ./bin/swctl single-metrics --name endpoint_cpm --ids
```
Comment (wu-sheng) [inline README.md]: All of those should use `--service-name=projectC`. What do you think? ID should be easy to use. right?

File: README.md [hunk 4]
```diff
@@ -77,83 +75,192 @@ This section covers all the available commands in SkyWalking CLI and their usage
...
+</details>
+
+<details>
+
+<summary>Query metrics single values for all endpoints of service of id 3</summary>
 
 ```shell
-go test -v ./...
+$ ./bin/swctl endpoint ls --service-id 3 | jq '.[] | [.id] | join(",")' | xargs ./bin/swctl single-metrics --name endpoint_cpm --end='2019-12-02 2137' --ids
+[{"id":"3","value":116}]
```
Comment (wu-sheng) [inline README.md]: And, you should list the output here, the graph I mean.

Comment (kezhenxu94) [reply README.md]: these two only has one value output, adding graph makes no sense, so the graph output is added above, which have multiple values

**Pull request merged**